### PR TITLE
[MODULAR] Adds 4 new guns to guncargo

### PR DESCRIPTION
**Pull request by RatFromTheJungle**
## About The Pull Request

Adds four new guns to the guncargo lists, none on their own are particularly strong(the mosin is kinda strong, but the combat shotgun still out damages it like 3-to-1), or even stronger then stuff elsewise present in the lists, but do help 'bulk it out'.
## How This Contributes To The Skyrat Roleplay Experience

Oh boy! Finally, a use for the dual tube!

## Changelog

:cl:
add: Adds the Dualtube Shotgun, Double barrel, detectives revolver, and the better mos-- brand new mosin to the guncargo companies.
/:cl:

### CODE REVIEW

**Comment by softcerv** [comment]
I am so glad to have the detective revolver back

File: modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm [hunk 1]
```diff
@@ -11,6 +11,11 @@
 	item_type = /obj/item/storage/belt/holster/thermal
 	lower_cost = CARGO_CRATE_VALUE * 8
 	upper_cost = CARGO_CRATE_VALUE * 12
+	
+/datum/armament_entry/cargo_gun/nanotrasen/pistol/revolver/detective
```
Comment (Zonespace27) [inline modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm:15]: make sure to add `COMPANY_NANOTRASEN` as a company bitflag to the gun itself

File: modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm [hunk 2]
```diff
@@ -20,10 +25,20 @@
 	lower_cost = CARGO_CRATE_VALUE * 8
 	upper_cost = CARGO_CRATE_VALUE * 12
 
+/datum/armament_entry/cargo_gun/nanotrasen/shotgun/doublebarrel
```
Comment (Zonespace27) [inline modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm:28]: ditto

File: modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm [hunk 3]
```diff
@@ -20,10 +25,20 @@
...
+	item_type = /obj/item/gun/ballistic/shotgun/doublebarrel
+	lower_cost = CARGO_CRATE_VALUE * 10
+	upper_cost = CARGO_CRATE_VALUE * 12
+	
 /datum/armament_entry/cargo_gun/nanotrasen/shotgun/combat
 	item_type = /obj/item/gun/ballistic/shotgun/automatic/combat
 	lower_cost = CARGO_CRATE_VALUE * 10
 	upper_cost = CARGO_CRATE_VALUE * 14
+	
+/datum/armament_entry/cargo_gun/nanotrasen/shotgun/dualtube
```
Comment (Zonespace27) [inline modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm:38]: ditto
also how does the dualtube differ from the doublebarrel

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm (hunk 3 above)
Comment (RatFromTheJungle) [reply modular_skyrat/modules/gun_cargo/code/armament_datums/nanotrasen.dm:38]: one is a doublebarrel, one is a shotgun(pump) with two seperate internal magazines that hold 3 each

File: modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm [hunk 4]
```diff
@@ -43,6 +43,11 @@
 	item_type = /obj/item/gun/ballistic/rifle/boltaction
 	lower_cost = CARGO_CRATE_VALUE * 5
 	upper_cost = CARGO_CRATE_VALUE * 6
+	
+/datum/armament_entry/cargo_gun/izhevsk/rifle/mosin/brandnew
+	item_type = /obj/item/gun/ballistic/rifle/boltaction/brand_new
```
Comment (Stalkeros2) [inline modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm:48]: There's literally no difference between the newer (brand-new) Mosin and the older (OG boltaction) one in terms of stats and sprites, except for the fact that the earlier can't be sawn-off. I'd advise either making a unique subtype with some "cewl" feature (like semi-auto/scoped Mosin? :flushed:) or not adding the brand-new one outright in favor of something else. A Nagant revolver possibly - it's hardly seen anywhere.

Comment (Zonespace27) [reply modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm:48]: Nagant revolver existed but was removed, please don't readd.

Comment (RatFromTheJungle) [reply modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm:48]: > There's literally no difference between the newer (brand-new) Mosin and the older (OG boltaction) one in terms of stats and sprites, except for the fact that the earlier can't be sawn-off. I'd advise either making a unique subtype with some "cewl" feature (like semi-auto/scoped Mosin? 😳) or not adding the brand-new one outright in favor of something else. A Nagant revolver possibly - it's hardly seen anywhere.

there is, bolt speed, brand_new's have a bolt delay of about 0.5s, while the default mosin has to wait 0.9 (essentially 1 second) between bolting (cycling) actions

**Review by Zonespace27 — APPROVED** [review]
Sure

File: modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm (hunk 4 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/gun_cargo/code/armament_datums/izhevsk.dm:48]: > there is, bolt speed, brand_new's have a bolt delay of about 0.5s, while the default mosin has to wait 0.9 (essentially 1 second) between bolting (cycling) actions

Oh epic.

**Pull request merged**